### PR TITLE
bluetooth: host: Add CS unit tests and helper API

### PR DESCRIPTION
**Pull request by olivier-le-sage**
Add CS unit tests and helper API

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

File: subsys/bluetooth/host/cs.c [hunk 1]
```diff
@@ -240,6 +240,24 @@ static void reset_reassembly_results(void)
 	memset(&reassembled_result, 0, sizeof(struct bt_conn_le_cs_subevent_result));
 }
 
+/** @brief Converts PCT to a pair of int16_t
+ *
+ */
+struct bt_le_cs_iq_sample bt_le_cs_parse_pct(const uint8_t pct[3])
```
Comment (olivier-le-sage) [inline subsys/bluetooth/host/cs.c:246]: @alwa-nordic fyki

File: tests/bluetooth/host/cs/bt_le_cs_parse_pct/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr COMPONENTS unittest REQUIRED HINTS $ENV{ZEPHYR_BASE})
+
+project(bt_step_data_parse)
```
Comment (theob-pro) [inline tests/bluetooth/host/cs/bt_le_cs_parse_pct/CMakeLists.txt:7]: The project should be `bt_parse_pct`, no?

File: tests/bluetooth/host/cs/mocks/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+add_library(mocks STATIC
```
Comment (theob-pro) [inline tests/bluetooth/host/cs/mocks/CMakeLists.txt:1]: Is it fine to not have license identifier there?

File: tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Dummy header file to avoid compiler errors
+ * intentionally left blank
+ */
```
Comment (theob-pro) [inline tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h:10]: Which compiler error? Is it something that should be reported and fixed later?

File: tests/bluetooth/host/cs/mocks/zephyr/linker/linker-defs.h [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+#ifdef DT_NODE_HAS_STATUS
+#undef DT_NODE_HAS_STATUS
+#endif
+
+#ifdef DT_FOREACH_OKAY_HELPER
+#undef DT_FOREACH_OKAY_HELPER
+#endif
+
+#define DT_FOREACH_OKAY_HELPER(x)
```
Comment (theob-pro) [inline tests/bluetooth/host/cs/mocks/zephyr/linker/linker-defs.h:17]: Also wondering about that. Should it be addressed by whoever maintain the unit_testing board?

**Review by theob-pro — DISMISSED** [review]

File: tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h (hunk 4 above)
Comment (olivier-le-sage) [reply tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h:10]: I copied these over from a different host unit test suite. I'm not sure if they're all really needed, let me check.

File: tests/bluetooth/host/cs/mocks/CMakeLists.txt (hunk 3 above)
Comment (olivier-le-sage) [reply tests/bluetooth/host/cs/mocks/CMakeLists.txt:3]: added

**Review by jhedberg — APPROVED** [review]

**Review by theob-pro — APPROVED** [review]

File: tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h (hunk 4 above)
Comment (olivier-le-sage) [reply tests/bluetooth/host/cs/mocks/zephyr/devicetree_generated.h]: seems not :)

**Pull request merged**